client/src/components/ContentNavigator/ContentModel.ts

```diff
@@ -322,19 +325,61 @@ export class ContentModel {
     }
   }
 
+  private getFileInfo(resourceId: string): {
+    etag: string;
+    lastModified: string;
+  } {
+    if (resourceId in this.fileTokenMaps) {
+      return this.fileTokenMaps[resourceId];
+    }
+    const now = new Date();
+    const timestamp = now.toUTCString();
+    return { etag: "", lastModified: timestamp };
+  }
+
   public async saveContentToUri(uri: Uri, content: string): Promise<void> {
     const resourceId = getResourceId(uri);
-    const res = await this.connection.put(resourceId + "/content", content, {
-      headers: {
-        "Content-Type": "text/plain",
-        "If-Match": this.fileTokenMaps[resourceId].etag,
-        "If-Unmodified-Since": this.fileTokenMaps[resourceId].lastModified,
-      },
-    });
-    this.fileTokenMaps[resourceId] = {
-      etag: res.headers.etag,
-      lastModified: res.headers["last-modified"],
+    const { etag, lastModified } = this.getFileInfo(resourceId);
+    const headers = {
+      "Content-Type": "text/plain",
+      "If-Unmodified-Since": lastModified,
     };
+    if (etag !== "") {
+      headers["If-Match"] = etag;
+    }
+    try {
+      const res = await this.connection.put(resourceId + "/content", content, {
+        headers,
+      });
+      this.fileTokenMaps[resourceId] = {
+        etag: res.headers.etag,
+        lastModified: res.headers["last-modified"],
+      };
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  public async associateFlowFile(
+    name: string,
+    uri: Uri,
+    parent: ContentItem,
+  ): Promise<string | undefined> {
+    try {
+      // create a studio session
+      const sessionId = await createStudioSession(this.connection);
+      // associate flow object
+      const flowUri = associateFlowObject(
```

Comment:
We _could_ just return here instead of assigning to a variable

Reply:
done in f5253ce014a5fec429456474725f717e0f2609a0